<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
    <title>ApiTest</title>
    <meta name="generator" content="HTMLTestRunner 0.9.1"/>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    
<style type="text/css" media="screen">
    body        { font-family: verdana, arial, helvetica, sans-serif; font-size: 80%; }
    table       { font-size: 100%; }
    pre         { white-space: pre-wrap;word-wrap: break-word; }

    /* -- heading ---------------------------------------------------------------------- */
    h1 {
        font-size: 16pt;
        color: gray;
    }
    .heading {
        margin-top: 0ex;
        margin-bottom: 1ex;
    }

    .heading .attribute {
        margin-top: 1ex;
        margin-bottom: 0;
    }

    .heading .description {
        margin-top: 2ex;
        margin-bottom: 3ex;
    }

    /* -- css div popup ------------------------------------------------------------------------ */
    a.popup_link {
    }

    a.popup_link:hover {
        color: red;
    }

    .popup_window {
        display: none;
        position: relative;
        left: 0px;
        top: 0px;
        /*border: solid #627173 1px; */
        padding: 10px;
        /*background-color: #E6E6D6; */
        font-family: "Lucida Console", "Courier New", Courier, monospace;
        text-align: left;
        font-size: 8pt;
        /* width: 500px;*/
    }

    }
    /* -- report ------------------------------------------------------------------------ */
    #show_detail_line {
        margin-top: 3ex;
        margin-bottom: 1ex;
    }
    #result_table {
        width: 99%;
    }
    #header_row {
        font-weight: bold;
        color: #303641;
        background-color: #ebebeb;
    }
    #total_row  { font-weight: bold; }
    .passClass  { background-color: #bdedbc; }
    .failClass  { background-color: #ffefa4; }
    .errorClass { background-color: #ffc9c9; }
    .passCase   { color: #6c6; }
    .failCase   { color: #FF6600; font-weight: bold; }
    .errorCase  { color: #c00; font-weight: bold; }
    .hiddenRow  { display: none; }
    .testcase   { margin-left: 2em; }


    /* -- ending ---------------------------------------------------------------------- */
    #ending {
    }

    #div_base {
                position:absolute;
                top:0%;
                left:5%;
                right:5%;
                width: auto;
                height: auto;
                margin: -15px 0 0 0;
    }
</style>

    <link href="http://cdn.bootcss.com/bootstrap/3.3.0/css/bootstrap.min.css" rel="stylesheet">
    <script type="text/javascript" src="https://cdn.bootcss.com/echarts/3.8.5/echarts.common.js"></script>
</head>
<body>
    <script language="javascript" type="text/javascript"><!--
    output_list = Array();

    /* level - 0:Summary; 1:Failed; 2:All */
    function showCase(level) {
        trs = document.getElementsByTagName("tr");
        for (var i = 0; i < trs.length; i++) {
            tr = trs[i];
            id = tr.id;
            if (id.substr(0,2) == 'ft') {
                if (level < 1) {
                    tr.className = 'hiddenRow';
                }
                else {
                    tr.className = '';
                }
            }
            if (id.substr(0,2) == 'pt') {
                if (level > 1) {
                    tr.className = '';
                }
                else {
                    tr.className = 'hiddenRow';
                }
            }
        }
    }


    function showClassDetail(cid, count) {
        var id_list = Array(count);
        var toHide = 1;
        for (var i = 0; i < count; i++) {
            tid0 = 't' + cid.substr(1) + '.' + (i+1);
            tid = 'f' + tid0;
            tr = document.getElementById(tid);
            if (!tr) {
                tid = 'p' + tid0;
                tr = document.getElementById(tid);
            }
            id_list[i] = tid;
            if (tr.className) {
                toHide = 0;
            }
        }
        for (var i = 0; i < count; i++) {
            tid = id_list[i];
            if (toHide) {
                document.getElementById('div_'+tid).style.display = 'none'
                document.getElementById(tid).className = 'hiddenRow';
            }
            else {
                document.getElementById(tid).className = '';
            }
        }
    }


    function showTestDetail(div_id){
        var details_div = document.getElementById(div_id)
        var displayState = details_div.style.display
        // alert(displayState)
        if (displayState != 'block' ) {
            displayState = 'block'
            details_div.style.display = 'block'
        }
        else {
            details_div.style.display = 'none'
        }
    }


    function html_escape(s) {
        s = s.replace(/&/g,'&amp;');
        s = s.replace(/</g,'&lt;');
        s = s.replace(/>/g,'&gt;');
        return s;
    }

    /* obsoleted by detail in <div>
    function showOutput(id, name) {
        var w = window.open("", //url
                        name,
                        "resizable,scrollbars,status,width=800,height=450");
        d = w.document;
        d.write("<pre>");
        d.write(html_escape(output_list[id]));
        d.write("\n");
        d.write("<a href='javascript:window.close()'>close</a>\n");
        d.write("</pre>\n");
        d.close();
    }
    */
    --></script>

    <div id="div_base">
        
    <div class='page-header'>
        <h1>ApiTest</h1>
    <p class='attribute'><strong>开始时间:</strong> 2018-09-16 18:14:41</p>
<p class='attribute'><strong>运行时长:</strong> 0:00:00.139454</p>
<p class='attribute'><strong>状态:</strong> 错误 2</p>

    </div>
    <div style="float: left;width:50%;"><p class='description'>测试报告</p></div>
    <div id="chart" style="width:50%;height:400px;float:left;"></div>

        
    <div class="btn-group btn-group-sm">
        <button class="btn btn-default" onclick='javascript:showCase(0)'>总结</button>
        <button class="btn btn-default" onclick='javascript:showCase(1)'>失败</button>
        <button class="btn btn-default" onclick='javascript:showCase(2)'>全部</button>
    </div>
    <p></p>
    <table id='result_table' class="table table-bordered">
        <colgroup>
            <col align='left' />
            <col align='right' />
            <col align='right' />
            <col align='right' />
            <col align='right' />
            <col align='right' />
        </colgroup>
        <tr id='header_row'>
            <td>测试套件/测试用例</td>
            <td>总数</td>
            <td>通过</td>
            <td>失败</td>
            <td>错误</td>
            <td>查看</td>
        </tr>
        
    <tr class='errorClass'>
        <td>user.test_user_login.TestUserLogin</td>
        <td>2</td>
        <td>0</td>
        <td>0</td>
        <td>2</td>
        <td><a href="javascript:showClassDetail('c1',2)">详情</a></td>
    </tr>

<tr id='ft1.1' class='none'>
    <td class='errorCase'><div class='testcase'>test_user_login_normal</div></td>
    <td colspan='5' align='center'>

    <!--css div popup start-->
    <a class="popup_link" onfocus='this.blur();' href="javascript:showTestDetail('div_ft1.1')" >
        错误</a>

    <div id='div_ft1.1' class="popup_window">
        <pre>
        ft1.1: Traceback (most recent call last):
  File "C:\Python3.6\lib\site-packages\urllib3\connection.py", line 171, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "C:\Python3.6\lib\site-packages\urllib3\util\connection.py", line 79, in create_connection
    raise err
  File "C:\Python3.6\lib\site-packages\urllib3\util\connection.py", line 69, in create_connection
    sock.connect(sa)
OSError: [WinError 10065] 套接字操作尝试一个无法连接的主机。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python3.6\lib\site-packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "C:\Python3.6\lib\site-packages\urllib3\connectionpool.py", line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "C:\Python3.6\lib\http\client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "C:\Python3.6\lib\http\client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "C:\Python3.6\lib\http\client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "C:\Python3.6\lib\http\client.py", line 1026, in _send_output
    self.send(msg)
  File "C:\Python3.6\lib\http\client.py", line 964, in send
    self.connect()
  File "C:\Python3.6\lib\site-packages\urllib3\connection.py", line 196, in connect
    conn = self._new_conn()
  File "C:\Python3.6\lib\site-packages\urllib3\connection.py", line 180, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: &lt;urllib3.connection.HTTPConnection object at 0x000002285162FF28&gt;: Failed to establish a new connection: [WinError 10065] 套接字操作尝试一个无法连接的主机。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python3.6\lib\site-packages\requests\adapters.py", line 445, in send
    timeout=timeout
  File "C:\Python3.6\lib\site-packages\urllib3\connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "C:\Python3.6\lib\site-packages\urllib3\util\retry.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='115.28.108.130', port=5000): Max retries exceeded with url: /api/user/login/ (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0x000002285162FF28&gt;: Failed to establish a new connection: [WinError 10065] 套接字操作尝试一个无法连接的主机。',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\sec\PycharmProjects\requests_demo4\testcase\user\test_user_login.py", line 26, in test_user_login_normal
    res = requests.post(url=url, data=json.loads(data))
  File "C:\Python3.6\lib\site-packages\requests\api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "C:\Python3.6\lib\site-packages\requests\api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Python3.6\lib\site-packages\requests\sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Python3.6\lib\site-packages\requests\sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "C:\Python3.6\lib\site-packages\requests\adapters.py", line 513, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='115.28.108.130', port=5000): Max retries exceeded with url: /api/user/login/ (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0x000002285162FF28&gt;: Failed to establish a new connection: [WinError 10065] 套接字操作尝试一个无法连接的主机。',))

        </pre>
    </div>
    <!--css div popup end-->

    </td>
</tr>

<tr id='ft1.2' class='none'>
    <td class='errorCase'><div class='testcase'>test_user_login_password_wrong</div></td>
    <td colspan='5' align='center'>

    <!--css div popup start-->
    <a class="popup_link" onfocus='this.blur();' href="javascript:showTestDetail('div_ft1.2')" >
        错误</a>

    <div id='div_ft1.2' class="popup_window">
        <pre>
        ft1.2: Traceback (most recent call last):
  File "C:\Python3.6\lib\site-packages\urllib3\connection.py", line 171, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "C:\Python3.6\lib\site-packages\urllib3\util\connection.py", line 79, in create_connection
    raise err
  File "C:\Python3.6\lib\site-packages\urllib3\util\connection.py", line 69, in create_connection
    sock.connect(sa)
OSError: [WinError 10065] 套接字操作尝试一个无法连接的主机。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python3.6\lib\site-packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "C:\Python3.6\lib\site-packages\urllib3\connectionpool.py", line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "C:\Python3.6\lib\http\client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "C:\Python3.6\lib\http\client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "C:\Python3.6\lib\http\client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "C:\Python3.6\lib\http\client.py", line 1026, in _send_output
    self.send(msg)
  File "C:\Python3.6\lib\http\client.py", line 964, in send
    self.connect()
  File "C:\Python3.6\lib\site-packages\urllib3\connection.py", line 196, in connect
    conn = self._new_conn()
  File "C:\Python3.6\lib\site-packages\urllib3\connection.py", line 180, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: &lt;urllib3.connection.HTTPConnection object at 0x00000228516E2278&gt;: Failed to establish a new connection: [WinError 10065] 套接字操作尝试一个无法连接的主机。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python3.6\lib\site-packages\requests\adapters.py", line 445, in send
    timeout=timeout
  File "C:\Python3.6\lib\site-packages\urllib3\connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "C:\Python3.6\lib\site-packages\urllib3\util\retry.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='115.28.108.130', port=5000): Max retries exceeded with url: /api/user/login/ (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0x00000228516E2278&gt;: Failed to establish a new connection: [WinError 10065] 套接字操作尝试一个无法连接的主机。',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\sec\PycharmProjects\requests_demo4\testcase\user\test_user_login.py", line 43, in test_user_login_password_wrong
    res = requests.post(url=url, data=data)
  File "C:\Python3.6\lib\site-packages\requests\api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "C:\Python3.6\lib\site-packages\requests\api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Python3.6\lib\site-packages\requests\sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Python3.6\lib\site-packages\requests\sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "C:\Python3.6\lib\site-packages\requests\adapters.py", line 513, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='115.28.108.130', port=5000): Max retries exceeded with url: /api/user/login/ (Caused by NewConnectionError('&lt;urllib3.connection.HTTPConnection object at 0x00000228516E2278&gt;: Failed to establish a new connection: [WinError 10065] 套接字操作尝试一个无法连接的主机。',))

        </pre>
    </div>
    <!--css div popup end-->

    </td>
</tr>

        <tr id='total_row'>
            <td>总计</td>
            <td>2</td>
            <td>0</td>
            <td>0</td>
            <td>2</td>
            <td>&nbsp;</td>
        </tr>
    </table>

        <div id='ending'>&nbsp;</div>
        
    <script type="text/javascript">
        // 基于准备好的dom，初始化echarts实例
        var myChart = echarts.init(document.getElementById('chart'));

        // 指定图表的配置项和数据
        var option = {
            title : {
                text: '测试执行情况',
                x:'center'
            },
            tooltip : {
                trigger: 'item',
                formatter: "{a} <br/>{b} : {c} ({d}%)"
            },
            legend: {
                orient: 'vertical',
                left: 'left',
                data: ['通过','失败','错误']
            },
            series : [
                {
                    name: '测试执行情况',
                    type: 'pie',
                    radius : '60%',
                    center: ['50%', '60%'],
                    data:[
                        {value:0, name:'通过'},
                        {value:0, name:'失败'},
                        {value:2, name:'错误'}
                    ],
                    itemStyle: {
                        emphasis: {
                            shadowBlur: 10,
                            shadowOffsetX: 0,
                            shadowColor: 'rgba(0, 0, 0, 0.5)'
                        }
                    }
                }
            ]
        };

        // 使用刚指定的配置项和数据显示图表。
        myChart.setOption(option);
    </script>
    
    </div>
</body>
</html>
